framework/src/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -525,10 +526,13 @@ private void applyDefaultsElement(final Element annotationScope, final Annotated
                 // but it doesn't work correctly and tests fail.
                 // (That whole @FromStubFile and @FromByteCode annotation
                 // logic should be replaced by something sensible.)
-            if ((ElementUtils.isElementFromByteCode(annotationScope) &&
+        	SourceChecker checker = atypeFactory.getContext().getChecker();
+        	if ((ElementUtils.isElementFromByteCode(annotationScope) &&
                     atypeFactory.declarationFromElement(annotationScope) == null &&
-                    !atypeFactory.isFromStubFile(annotationScope)) ||
-                    !annotatedForThisChecker
+                    !atypeFactory.isFromStubFile(annotationScope) &&
+                    checker.hasOption("safeDefaultsForUnannotatedBytecode")) || // TODO: make safeDefaultsForUnannotatedBytecode the default behavior.
```

Comment:
I think the code would be clearer with the hasOption check first in each clause.
Also, the TODO here feels out of place.  This code won't change when the default is changed.